src/kernel/pmm.zig

```diff
@@ -0,0 +1,175 @@
+const is_test = @import("builtin").is_test;
+const std = @import("std");
+const build_options = @import("build_options");
+const mock_path = build_options.mock_path;
+const arch = @import("arch.zig").internals;
+const MemProfile = (if (is_test) @import(mock_path ++ "mem_mock.zig") else @import("mem.zig")).MemProfile;
+const testing = std.testing;
+const panic = @import("panic.zig").panic;
+const log = @import("log.zig");
+const MEMORY_AVAILABLE = @import("multiboot.zig").MULTIBOOT_MEMORY_AVAILABLE;
+const Bitmap = @import("bitmap.zig").Bitmap;
+
+const PmmBitmap = Bitmap(u32);
+
+/// The possible errors thrown by bitmap functions
+const Error = error{
+    /// The address given hasn't been allocated
+    NotAllocated,
+};
+
+/// The size of memory associated with each bitmap entry
+const BLOCK_SIZE = arch.MEMORY_BLOCK_SIZE;
+
+var bitmap: PmmBitmap = undefined;
+
+///
+/// Set the bitmap entry for an address as occupied
+///
+/// Arguments:
+///     IN addr: usize - The address.
+///
+/// Error: PmmBitmap.Error.
+///     *: See PmmBitmap.setEntry.
+///
+fn setAddr(addr: usize) PmmBitmap.Error!void {
+    try bitmap.setEntry(@intCast(u32, addr / BLOCK_SIZE));
+}
+
+///
+/// Check if an address is set as occupied.
+///
+/// Arguments:
+///     IN addr: usize - The address to check.
+///
+/// Return: True if occupied, else false.
+///
+/// Error: PmmBitmap.Error.
+///     *: See PmmBitmap.setEntry.
+///
+fn isSet(addr: usize) PmmBitmap.Error!bool {
```

Comment:
Unit test

Reply:
Done. I merged it with the alloc and free tests as testing isSet requires them to be called anyway.